zerver/models.py

```diff
@@ -390,33 +422,33 @@ def get_realm_filters_cache_key(domain):
     return u'all_realm_filters:%s' % (domain,)
 
 # We have a per-process cache to avoid doing 1000 remote cache queries during page load
-per_request_realm_filters_cache = {} # type: Dict[text_type, List[Tuple[text_type, text_type]]]
+per_request_realm_filters_cache = {} # type: Dict[text_type, List[Tuple[text_type, text_type, int]]]
 
 def domain_in_local_realm_filters_cache(domain):
     # type: (text_type) -> bool
     return domain in per_request_realm_filters_cache
 
 def realm_filters_for_domain(domain):
-    # type: (text_type) -> List[Tuple[text_type, text_type]]
+    # type: (text_type) -> List[Tuple[text_type, text_type, int]]
     domain = domain.lower()
     if not domain_in_local_realm_filters_cache(domain):
         per_request_realm_filters_cache[domain] = realm_filters_for_domain_remote_cache(domain)
     return per_request_realm_filters_cache[domain]
 
 @cache_with_key(get_realm_filters_cache_key, timeout=3600*24*7)
 def realm_filters_for_domain_remote_cache(domain):
-    # type: (text_type) -> List[Tuple[text_type, text_type]]
+    # type: (text_type) -> List[Tuple[text_type, text_type, int]]
     filters = []
     for realm_filter in RealmFilter.objects.filter(realm=get_realm(domain)):
-       filters.append((realm_filter.pattern, realm_filter.url_format_string))
+        filters.append((realm_filter.pattern, realm_filter.url_format_string, realm_filter.pk))
 
     return filters
 
 def all_realm_filters():
-    # type: () -> Dict[text_type, List[Tuple[text_type, text_type]]]
-    filters = defaultdict(list) # type: Dict[text_type, List[Tuple[text_type, text_type]]]
+    # type: () -> Dict[text_type, List[Tuple[text_type, text_type, int]]]
+    filters = defaultdict(list) # type: Dict[text_type, List[Tuple[text_type, text_type, int]]]
     for realm_filter in RealmFilter.objects.all():
-       filters[realm_filter.realm.domain].append((realm_filter.pattern, realm_filter.url_format_string))
+        filters[realm_filter.realm.domain].append((realm_filter.pattern, realm_filter.url_format_string, realm_filter.pk))
```

Comment:
use `realm_filter.id`, not `realm_filter.pk` -- as a project, we prefer to be explicit about which field is the primary key being used.